### PR TITLE
Create a plugin Avalonia framework POC

### PR DESCRIPTION
**Pull request by nick-polyak-ms**
Here is the link to the corresponding Issue https://github.com/morganstanley/ComposeUI/issues/60. 
Please, note - there are not changes to src - all the changes only modify code and prototypes within Tryouts folder.
The main solution  to run for this POC is MorganStanley.ComposeUI.Prototypes.ThemeChangerPrototype.sln

### CODE REVIEW

**Comment by ZKRobi** [comment]
None of the files seem to have the license header, please add them

File: Tryouts/Core/Utilities/ViewModelBase.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using System.ComponentModel;
+
+namespace MorganStanley.ComposeUI.Tryouts.Core.Utilities
+{
+    public class ViewModelBase : INotifyPropertyChanged
+    {
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        public void OnPropertyChanged(string propName)
```
Comment (ztanczos) [inline Tryouts/Core/Utilities/ViewModelBase.cs:9]: Nitpick: should be protected

File: Tryouts/Plugins/Services/CommunicationsServices/GrpcCommunicationsService/GrpcCommunicationsSeriviceImpl.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
+﻿using Grpc.Core;
+using MorganStanley.ComposeUI.Tryouts.Core.Abstractions;
+using NP.Utilities.Attributes;
+using Subscriptions;
+
+namespace MorganStanley.ComposeUI.Services.CommunicationsServices.GrpcCommunicationsService
+{
+    [Implements(typeof(ICommunicationService), isSingleton:true)]
+    public class GrpcCommunicationsSeriviceImpl : ICommunicationService
```
Comment (ztanczos) [inline Tryouts/Plugins/Services/CommunicationsServices/GrpcCommunicationsService/GrpcCommunicationsSeriviceImpl.cs:9]: Nitpick: typo in class + file name

File: Tryouts/Plugins/Services/CommunicationsServices/GrpcCommunicationsService/SubscriptionClientImpl.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+{
+    [Implements(typeof(ISubscriptionClient), isSingleton:true)]
+    public class SubscriptionClientImpl : ISubscriptionClient
+    {
+        private SubscriptionsServiceClient _client;
+        CancellationTokenSource _cts = new CancellationTokenSource();
+
+        public async Task Connect(string host, int port)
+        {
+            await Task.Delay(5000);
```
Comment (ztanczos) [inline Tryouts/Plugins/Services/CommunicationsServices/GrpcCommunicationsService/SubscriptionClientImpl.cs:19]: Can you please explain why do you need this delay?

File: Tryouts/Core/Utilities/ViewModelBase.cs (hunk 1 above)
Comment (nick-polyak-ms) [reply Tryouts/Core/Utilities/ViewModelBase.cs]: changed

**Comment by nick-polyak-ms** [comment]
> None of the files seem to have the license header, please add them

Added license headers to all the relevant files.

File: Tryouts/Plugins/Services/CommunicationsServices/GrpcCommunicationsService/SubscriptionClientImpl.cs (hunk 3 above)
Comment (nick-polyak-ms) [reply Tryouts/Plugins/Services/CommunicationsServices/GrpcCommunicationsService/SubscriptionClientImpl.cs:30]: I use this delay to let the server start. Of course, we have to come up with a better way of notifying the clients that the server is up, but for the tryouts I think it is ok. We won't move it to src until this problem is resolved.

**Pull request merged**